modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesSubscriptionManager.java

```diff
@@ -103,35 +104,44 @@ public void addSubscription(LocalSubscription localSubscription) throws AndesExc
                     matchingSubscriptions.add(existingSubscription);
                 }
             }
-
-            //there is one inactive subscription (to represent inactive subscription). We need delete it
-            if(matchingSubscriptions.size() == 1) {
-                AndesSubscription matchingSubscription = matchingSubscriptions.get(0);
-                if(!matchingSubscription.hasExternalSubscriptions()) {
-                    //delete the above subscription (only if subscription is activated from a different node -
-                    // decided looking at subscription ID and the subscribed node)
-                    if(!matchingSubscription.getSubscriptionID().equals(localSubscription.getSubscriptionID())
-                             || !matchingSubscription.getSubscribedNode().equals(localSubscription.getSubscribedNode())) {
-                        mockSubscription = convertClusterSubscriptionToMockLocalSubscription
-                                (matchingSubscription);
-                        mockSubscription.close();
-                    } else {
-                        subscriptionStore.updateLocalSubscriptionSubscription(localSubscription);
-                        durableTopicSubFoundAndUpdated = true;
-                    }
-
-                } else {
+            Iterator<AndesSubscription> subIterator = matchingSubscriptions.iterator();
+            //See whether there are existing matching active subscriptions
+            while (subIterator.hasNext()) {
```

Comment:
Can't we do this in the above for loop where matchingSubscriptions are populated?